src/xb-builder.c

```diff
@@ -852,6 +852,7 @@ xb_builder_compile(XbBuilder *self,
 		for (guint i = 0; i < nodes_to_destroy->len; i++) {
 			XbBuilderNode *bn = g_ptr_array_index(nodes_to_destroy, i);
 			xb_builder_node_unlink(bn);
+			g_object_unref(bn);
```

Comment:
could we define nodes_to_destroy with `g_ptr_array_new_with_free_func((GDestroyNotify)g_object_unref);` to avoid this?

Reply:
Of course!

Reply:
(just testing it with gnome software before sending the patch)

Reply:
Done.